code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -108,8 +108,7 @@
 
 /obj/item/weapon/melee/energy/sword/dropped(var/mob/user)
 	..()
-	if(!istype(loc,/mob))
```

Comment:
this check is no longer needed since now dropped will never fire until the object leaves the mob